plutarch-orphanage/src/PlutusLedgerApi/V2/Orphans.hs

```diff
@@ -0,0 +1,1271 @@
+import Data.Coerce (coerce)
+import Data.Set qualified as Set
+import Data.Word (Word32)
+import PlutusCore.Data qualified as PLC
+import PlutusLedgerApi.QuickCheck.Utils (
+  fromAsWord64,
+  unSizedByteString,
+ )
+import PlutusLedgerApi.V1.Interval qualified as Interval
+import PlutusLedgerApi.V2 qualified as PLA
+import PlutusTx.AssocMap qualified as AssocMap
+import PlutusTx.Builtins qualified as Builtins
+import PlutusTx.Prelude qualified as PlutusTx
+import Test.QuickCheck (
+  Arbitrary (arbitrary, shrink),
+  Arbitrary1 (liftArbitrary, liftShrink),
+  CoArbitrary (coarbitrary),
+  Function (function),
+  Gen,
+  NonNegative (NonNegative),
+  chooseBoundedIntegral,
+  chooseInt,
+  frequency,
+  functionMap,
+  getNonNegative,
+  oneof,
+  resize,
+  scale,
+  sized,
+  variant,
+  vectorOf,
+ )
+import Test.QuickCheck.Instances.ByteString ()
+
+-- | @since 1.0.0
+instance Arbitrary PlutusTx.BuiltinByteString where
+  {-# INLINEABLE arbitrary #-}
+  arbitrary = PlutusTx.toBuiltin @ByteString <$> arbitrary
+  {-# INLINEABLE shrink #-}
+  shrink = fmap (PlutusTx.toBuiltin @ByteString) . shrink . PlutusTx.fromBuiltin
+
+-- | @since 1.0.0
+instance CoArbitrary PlutusTx.BuiltinByteString where
+  {-# INLINEABLE coarbitrary #-}
+  coarbitrary = coarbitrary . PlutusTx.fromBuiltin
+
+-- | @since 1.0.0
+instance Function PlutusTx.BuiltinByteString where
+  {-# INLINEABLE function #-}
+  function = functionMap PlutusTx.fromBuiltin (PlutusTx.toBuiltin @ByteString)
+
+{- | Base-16 encoded bytestring.
+
+@since 1.0.0
+-}
+instance Arbitrary PLA.LedgerBytes where
+  {-# INLINEABLE arbitrary #-}
+  arbitrary = PLA.LedgerBytes . PlutusTx.toBuiltin @ByteString . Base16.encode <$> arbitrary
+  {-# INLINEABLE shrink #-}
+  -- We use the lenient strategy here, as it's not possible for us to ever end
+  -- up with anything that wasn't base-16 to begin with.
+  shrink =
+    fmap (PLA.LedgerBytes . PlutusTx.toBuiltin @ByteString . Base16.encode)
+      . shrink
+      . Base16.decodeLenient
+      . PlutusTx.fromBuiltin
+      . coerce @PLA.LedgerBytes @PlutusTx.BuiltinByteString
+
+-- | @since 1.0.0
+deriving via PlutusTx.BuiltinByteString instance CoArbitrary PLA.LedgerBytes
+
+-- | @since 1.0.0
+instance Function PLA.LedgerBytes where
+  {-# INLINEABLE function #-}
+  function = functionMap coerce PLA.LedgerBytes
+
+{- | BLAKE2b-244 hash. This does not shrink.
+
+@since 1.0.0
+-}
+instance Arbitrary PLA.PubKeyHash where
+  {-# INLINEABLE arbitrary #-}
+  arbitrary =
+    PLA.PubKeyHash . PlutusTx.toBuiltin @ByteString . unSizedByteString @28 <$> arbitrary
+
+-- | @since 1.0.0
+deriving via PlutusTx.BuiltinByteString instance CoArbitrary PLA.PubKeyHash
+
+-- | @since 1.0.0
+instance Function PLA.PubKeyHash where
+  {-# INLINEABLE function #-}
+  function = functionMap coerce PLA.PubKeyHash
+
+{- | BLAKE2b-244 hash. This does not shrink.
+
+@since 1.0.0
+-}
+deriving via PLA.PubKeyHash instance Arbitrary PLA.ScriptHash
```

Comment:
Consider having a Blake224Hash type and derive instances from there for PublicKeyHash and ScriptHash?